Source/SwiftLintBuiltInRules/Rules/Idiomatic/NoEmptyBlockRule.swift

```diff
@@ -102,13 +103,36 @@ private extension NoEmptyBlockRule {
             // No need to show a warning since Empty Block of `guard` is compile error.
             guard node.parent?.kind != .guardStmt else { return }
 
+            if let codeBlockType = node.codeBlockType, configuration.enabledBlockTypes.contains(codeBlockType) {
+                validate(node: node)
+            }
+        }
+
+        func validate(node: CodeBlockSyntax) {
             guard node.statements.isEmpty,
                   !node.leftBrace.trailingTrivia.containsComments,
                   !node.rightBrace.leadingTrivia.containsComments else {
                 return
             }
-
             violations.append(node.leftBrace.positionAfterSkippingLeadingTrivia)
         }
     }
 }
+
+private extension CodeBlockSyntax {
+    var codeBlockType: NoEmptyBlockConfiguration.CodeBlockType? {
+        guard let kind = self.parent?.kind else { return nil }
+        switch kind {
```

Comment:
```suggestion
        switch parent?.kind {
```

Then you can make use of a `switch` expression and leave out the `return` keywords. 😉